src/db/spanner/schema.ddl

```diff
@@ -41,58 +44,40 @@ INTERLEAVE IN user_collections;
 INTERLEAVE IN user_collections;
 
 CREATE TABLE collections (
-  collection_id INT64  NOT NULL,
-  name STRING(32)      NOT NULL,
+    collection_id INT64 NOT NULL,
+    name STRING(32)     NOT NULL,
 ) PRIMARY KEY(collection_id);
 
     CREATE UNIQUE INDEX CollectionName
         ON collections(name);
 
 CREATE TABLE batches (
-  fxa_uid STRING(MAX)  NOT NULL,
-  fxa_kid STRING(MAX)  NOT NULL,
-  collection_id INT64  NOT NULL,
-  batch_id STRING(MAX) NOT NULL,
-  expiry TIMESTAMP     NOT NULL,
+    fxa_uid STRING(MAX)  NOT NULL,
+    fxa_kid STRING(MAX)  NOT NULL,
+    collection_id INT64  NOT NULL,
+    batch_id STRING(MAX) NOT NULL,
+    expiry TIMESTAMP     NOT NULL,
 )    PRIMARY KEY(fxa_uid, fxa_kid, collection_id, batch_id),
   INTERLEAVE IN PARENT user_collections ON DELETE CASCADE;
 
-    CREATE INDEX BatchExpiry
-        ON batches(expiry);
+    CREATE INDEX BatchExpireId
+        ON batches(fxa_uid, fxa_kid, collection_id, expiry),
+    INTERLEAVE IN user_collections;
 
 CREATE TABLE batch_bsos (
-  fxa_uid STRING(MAX)      NOT NULL,
-  fxa_kid STRING(MAX)      NOT NULL,
-  collection_id INT64      NOT NULL,
-  batch_id STRING(MAX)     NOT NULL,
-  batch_bso_id STRING(MAX) NOT NULL,
-
-  sortindex INT64,
-  payload STRING(MAX),
-  ttl INT64,
+    fxa_uid STRING(MAX)     NOT NULL,
+    fxa_kid STRING(MAX)     NOT NULL,
+    collection_id INT64     NOT NULL,
+    batch_id STRING(MAX)    NOT NULL,
+    batch_bso_id STRING(64) NOT NULL,
+
+    sortindex INT64,
+    payload STRING(MAX),
+    ttl INT64,
 )    PRIMARY KEY(fxa_uid, fxa_kid, collection_id, batch_id, batch_bso_id),
   INTERLEAVE IN PARENT batches ON DELETE CASCADE;
 
 -- batch_bsos' bso fields are nullable as the batch upload may or may
 -- not set each individual field of each item. Also note that there's
 -- no "modified" column because the modification timestamp gets set on
 -- batch commit.
-
```

Comment:
In the interest of over-documenting, we might want to add a comment here like:

```
-- *NOTE*: 
-- Newly created Spanner instances should pre-populate the `collections` table by 
-- running the content of `insert_standard_collections.sql `
```

(Again, trying to be kind to the future person, who's in a panic.)